pkg/eigenState/operatorDirectedRewardSubmissions/operatorDirectedRewardSubmissions.go

```diff
@@ -0,0 +1,334 @@
+	stateChanges, sortedBlockNumbers := odrs.GetStateTransitions()
+
+	for _, blockNumber := range sortedBlockNumbers {
+		if log.BlockNumber >= blockNumber {
+			odrs.logger.Sugar().Debugw("Handling state change", zap.Uint64("blockNumber", log.BlockNumber))
+
+			change, err := stateChanges[blockNumber](log)
+			if err != nil {
+				return nil, err
+			}
+			if change == nil {
+				return nil, nil
+			}
+			return change, nil
+		}
+	}
+	return nil, nil
+}
+
+// prepareState prepares the state for commit by adding the new state to the existing state.
+func (odrs *OperatorDirectedRewardSubmissionsModel) prepareState(blockNumber uint64) ([]*OperatorDirectedRewardSubmission, error) {
+	accumulatedState, ok := odrs.stateAccumulator[blockNumber]
+	if !ok {
+		err := xerrors.Errorf("No accumulated state found for block %d", blockNumber)
+		odrs.logger.Sugar().Errorw(err.Error(), zap.Error(err), zap.Uint64("blockNumber", blockNumber))
+		return nil, err
+	}
+
+	recordsToInsert := make([]*OperatorDirectedRewardSubmission, 0)
+	for _, submission := range accumulatedState {
+		recordsToInsert = append(recordsToInsert, submission)
+	}
+	return recordsToInsert, nil
+}
+
+// CommitFinalState commits the final state for the given block number.
+func (odrs *OperatorDirectedRewardSubmissionsModel) CommitFinalState(blockNumber uint64) error {
+	recordsToInsert, err := odrs.prepareState(blockNumber)
+	if err != nil {
+		return err
+	}
+
+	if len(recordsToInsert) > 0 {
+		for _, record := range recordsToInsert {
+			res := odrs.DB.Model(&OperatorDirectedRewardSubmission{}).Clauses(clause.Returning{}).Create(&record)
+			if res.Error != nil {
+				odrs.logger.Sugar().Errorw("Failed to insert records", zap.Error(res.Error))
+				return res.Error
+			}
+		}
+	}
+	return nil
+}
+
+// GenerateStateRoot generates the state root for the given block number using the results of the state changes.
+func (odrs *OperatorDirectedRewardSubmissionsModel) GenerateStateRoot(blockNumber uint64) (types.StateRoot, error) {
+	inserts, err := odrs.prepareState(blockNumber)
+	if err != nil {
+		return "", err
+	}
+
+	inputs := odrs.sortValuesForMerkleTree(inserts)
+
+	if len(inputs) == 0 {
+		return "", nil
+	}
+
+	fullTree, err := odrs.MerkleizeState(blockNumber, inputs)
+	if err != nil {
+		odrs.logger.Sugar().Errorw("Failed to create merkle tree",
+			zap.Error(err),
+			zap.Uint64("blockNumber", blockNumber),
+			zap.Any("inputs", inputs),
+		)
+		return "", err
+	}
+	return types.StateRoot(utils.ConvertBytesToString(fullTree.Root())), nil
+}
+
+func (odrs *OperatorDirectedRewardSubmissionsModel) sortValuesForMerkleTree(submissions []*OperatorDirectedRewardSubmission) []*base.MerkleTreeInput {
+	inputs := make([]*base.MerkleTreeInput, 0)
+	for _, submission := range submissions {
+		slotID := NewSlotID(submission.TransactionHash, submission.LogIndex, submission.RewardHash, submission.StrategyIndex, submission.OperatorIndex)
+		value := "added"
```

Comment:
probably makes sense to use `submission.Amount` here rather than "added"

Reply:
What is the purpose of this value field?

Reply:
Done in [997e1b3](https://github.com/Layr-Labs/sidecar/pull/112/commits/997e1b36c45bcf628de0e2e26696f535d57fff24). I've updated it with what I think is a better merkle leaf value.